### ISSUE TITLE
Inputs extend outside of custom procedures modal

### ISSUE THREAD

**Issue by kchadha** [LLK/scratch-gui#997]
### Expected Behavior

Things should not extend outside of modals.

### Actual Behavior

![image](https://user-images.githubusercontent.com/1786240/33568021-0b00d9e2-d8f3-11e7-9084-e1a8cade9bd8.png)

### Steps to Reproduce

Create a new procedure block with many long inputs.

### Operating System and Browser

_e.g. Mac OS 10.11.6 Safari 10.0_

**Comment by mrjacobbloom** [comment]
This relates to LLK/scratch-blocks#1255 and likewise I'm not sure what the least-bad response should be--at least the mini-workspace is scrollable though

**Comment by paulkaplan** [comment]
@mrjacobbloom good point, that issue has some interesting discussion. I think a solution that keeps the focused input within the view will get 99% of the way there, the final 1% being what happens when a single input is wider than the entire view. And like that linked issues states, there isn't really a sound solution for that case, all we can do is make sure it doesn't break other things in the UI. 

This PR https://github.com/LLK/scratch-gui/pull/1001 is a first pass at keeping the focused in view. I'm struggling with a full solution, because I do not think there are scratch-blocks events for when the field is focused.

**Comment by thisandagain** [comment]
Agreed that https://github.com/LLK/scratch-gui/pull/1001 will resolve 99% of this issue. Backlogging for now.
